src/main/kotlin/nomic/data/repositories/games/GameSpecificationFactory.kt

```diff
@@ -0,0 +1,20 @@
+package nomic.data.repositories.games
+
+import nomic.domain.entities.EndUser
+import nomic.domain.games.IGameSpecificationFactory
+import org.springframework.stereotype.Service
+
+@Service
+class GameSpecificationFactory : IGameSpecificationFactory {
+    override fun filterByUser(user: EndUser): IGameSpecification {
+        return FilterGamesByUserSpecification(user)
+    }
+
+    override fun paginate(size: UInt, offset: UInt): IGameSpecification {
+        return PaginateGamesSpecification(size, offset)
+    }
+
+    override fun sortByMostRecentFirst(): IGameSpecification {
+        return SortGamesByMostRecentFirstSpecification()
+    }
+}
```

Comment:
Honestly, I wish you had discussed with the team whether a gameSpecificationFactory abstraction was something that needed to be considered in this user story as this could have saved you a lot of unnecessary work.